src/gcp_scanner/crawler/crawler_factory.py

```diff
@@ -23,28 +25,26 @@
 from gcp_scanner.crawler.machine_images_crawler import ComputeMachineImagesCrawler
 
 
+service_crawler_map = {
```

Comment:
hmm @sudiptob2 anyway we can measure impact of that? If it is a minor memory uptick then I am fine with that given we have better readability.

Reply:
>  If these objects are lazy instantiated then I don't have any problem with this method.

Nope, by default python does not `lazy evaluate` functions. We have to define them explicitly.

Another approach that I found while researching was to reference these functions instead of creating an instance, i.e., 

```python
service_crawler_map = {
  "app_services": AppServicesCrawler,
}
```
**v/s**
```python
service_crawler_map = {
  "app_services": AppServicesCrawler(),
}
```
and later call them as necessary. Example: `service_crawler_map[name]()`

This reference also has a memory impact. But, compared to `func()`(instance), `func`(reference) its memory impact is negligible.

Reply:
@peb-peb both of this is fine given impact is small as you mentioned. Thanks for the extra research.